mongo/options/clientencryptionoptions.go

```diff
@@ -43,7 +80,54 @@ func MergeClientEncryptionOptions(opts ...*ClientEncryptionOptions) *ClientEncry
 		if opt.KmsProviders != nil {
 			ceo.KmsProviders = opt.KmsProviders
 		}
+		if opt.TLSConfig != nil {
+			ceo.TLSConfig = opt.TLSConfig
+		}
 	}
 
 	return ceo
 }
+
+// Used by both ClientEncryptionOptions.SetTLSOptions() and AutoEncryptionOptions.SetTLSOptions()
+func applyTLSOptions(tlsOpts map[string]map[string]interface{}, tlsConfigs map[string]*tls.Config) (map[string]*tls.Config, error) {
```

Comment:
I suggest removing the `tlsConfigs` argument and have `applyTLSOptions` create the initial empty map.